server/src/instant/db/transaction.clj

```diff
@@ -140,6 +140,37 @@
      app-id
      [{:message (format "You can't make updates to this app.")}])))
 
+(defn enforce-on-deletes
+  "Unoptimized on-delete helper that will delete entities that have
+   on-delete=:cascade. Only used for system attributes right now.
+   Before releasing as a feature, this should probably happen inside of
+   the `delete-multi`.
+
+   This approach is not ideal because there's no bound on the number of
+   times you recur. We probably want to prevent cycles on the on-delete
+   (e.g. if `a` has on-delete for `b`, then `b` can't have on-delete
+   for `a`), so that the delete query can be statically defined in
+   triple-model/delete-multi!"
+  ([conn attrs app-id results]
+   (enforce-on-deletes conn attrs app-id results (:delete-entity results)))
+  ([conn attrs app-id results last-delete-entities]
+   (if-let [deleted-triples (seq last-delete-entities)]
+     (let [deletes (keep (fn [{:keys [triples/attr_id triples/entity_id]}]
+                           (let [attr (attr-model/seek-by-id attr_id attrs)]
+                             (when (and (= :ref (:value-type attr))
+                                        (= :cascade (:on-delete attr)))
```

Comment:
Only `:cascade` for now, but we could also support `:restrict`, which would throw an error if you had more items to delete.

Our default is technically `:setnull`, so we could make that the default on the attr.